learning_journal_basic/learning_journal_basic/tests.py

```diff
@@ -1,29 +1,157 @@
+
+def test_list_view(dummy_request, new_session):
+    """Test list view returns my entry titles."""
+    from .views.default import list_view
+    new_session.add(MyEntry(
+        title="testing title",
+        creation_date=datetime.utcnow(),
+        blog_entry="test blog entry"
+    ))
+    new_session.flush()
+    response = list_view(dummy_request)
+    assert response['entries'][0].title == "testing title"
+
+def test_detail_view(new_session):
+    """Test detail view return blog entry."""
+    from .views.default import detail_view
+    req = testing.DummyRequest(dbsession=new_session)
+    new_session.add(MyEntry(
+        title="testing title",
+        creation_date=datetime.utcnow(),
+        blog_entry="test blog entry"
+    ))
+    new_session.flush()
+    req.matchdict = {"id": 1}
+    response = detail_view(req)
+    assert response['entry'].blog_entry == "test blog entry"
+
+def test_create_view(dummy_request):
+    """Test create view returning an empty dictionary."""
+    from .views.default  import create_view
+    response = create_view(dummy_request)
+    assert response == {}
+
+def test_update_view(dummy_request, new_session):
+    """Test update view return blog entry."""
+    from .views.default import update_view
+    new_session.add(MyEntry(
+        title="this is my update",
+        creation_date=datetime.utcnow(),
+        blog_entry="update sexting"
+    ))
+    new_session.flush()
+    dummy_request.matchdict = {"id": 1}
+    response = update_view(dummy_request)
+    assert response['entry'].blog_entry == "update sexting"
+
+@pytest.fixture(scope="session")
+def testapp(request):
+    from webtest import TestApp
+    from learning_journal_basic import main
+    app = main({}, **{"sqlalchemy.url": "sqlite:///:memory:"})
+    testapp = TestApp(app)
+
+    SessionFactory = app.registry["dbsession_factory"]
+    engine = SessionFactory().bind
+    Base.metadata.create_all(bind=engine)
+
+    def tearDown():
+        Base.metadata.drop_all(bind=engine)
+
+    request.addfinalizer(tearDown)
+    return testapp
+
+@pytest.fixture(scope="session")
+def fill_db(testapp):
+    SessionFactory = testapp.app.registry["dbsession_factory"]
+    with transaction.manager:
+        dbsession = get_tm_session(SessionFactory, transaction.manager)
+        dbsession.add_all(ENTRIES)
+
+    return dbsession
+
+@pytest.fixture(scope="function")
+def new_session(testapp, request):
+    SessionFactory = testapp.app.registry["dbsession_factory"]
+    with transaction.manager:
+        session = get_tm_session(SessionFactory, transaction.manager)
+    def teardown():
+        transaction.abort()
+    request.addfinalizer(teardown)
+    return session
 
-    def test_my_view(self):
-        from .views import my_view
-        request = testing.DummyRequest()
-        info = my_view(request)
-        self.assertEqual(info['project'], 'learning_journal_basic')
+#----------------------  functional tests --------------------
```

Comment:
Consider moving the setup fixture for functional tests inside this section to make it more readable.